File: pinyougouparent/pinyougou_sellergoods_interface/src/main/java/com/pinyougou/sellergoods/service/BrandService.java
package com.pinyougou.sellergoods.service;

import com.pinyougou.pojo.TbBrand;
import entity.PageResult;

import java.util.List;

//品牌服务接口

public interface BrandService {
    //查询所有品牌
    List<TbBrand> findAll();

    //分页查询
    PageResult findPage(int pageNum, int pageSize);//pageNum为当前页码，pageSize为每页显示的条数

    //增加品牌
    void add(TbBrand brand);

    //修改品牌
    void update(TbBrand brand);

    //根据id查询
    TbBrand findOne(long id);

    //批量删除
    void delete(long[] ids);

    //复杂条件查询，并完成分页
    PageResult search(TbBrand brand, int pageNum, int pageSize);
}

File: pinyougouparent/pinyougou_manager_web/src/main/webapp/js/service/brandService.js
//定义服务
app.service("brandService", function ($http) {
    //查询所有品牌
    this.findAll = function () {
        return $http.get('../brand/findAll.do')
    }

    //分页查询品牌
    this.findPage = function (page, rows) {
        return $http.get('../brand/findPage.do?page=' + page + '&size=' + rows);
    }

    //查询单个品牌
    this.findOne = function (id) {
        return $http.get('../brand/findOne.do?id=' + id);
    }

    //批量删除
    this.del = function (selectedIds) {
        return $http.get('../brand/delete.do?ids=' + selectedIds);
    }

    //添加品牌
    this.add = function (entity) {
        return $http.post('../brand/add.do', entity);
    }

    //修改品牌
    this.update = function (entity) {
        return $http.post('../brand/update.do', entity);
    }

    //负责条件查询
    this.search = function (page, rows , searchEntity) {
        return $http.post('../brand/search.do?page=' + page + '&size=' + rows, searchEntity);
    }

})